src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -65,6 +74,53 @@ public List<ExtensionRestHandler> getExtensionRestHandlers() {
             );
     }
 
+    @Override
+    public Collection<Object> createComponents(ExtensionsRunner runner) {
+
+        SDKRestClient sdkRestClient = getRestClient();
```

Comment:
It's subtly hidden in the getter, but this needs to be closed when we're done with it, so we should make an effort to keep the same instance everywhere in the extension.  Accordingly you should return this client as one of the objects so it gets injected, and can be closed when we want to shut down the extension.